esmvalcore/cmor/table.py

```diff
@@ -124,9 +124,17 @@ def read_cmor_tables(cfg_developer: Optional[Path] = None) -> None:
     ----------
     cfg_developer:
         Path to config-developer.yml file.
+
+    Raises
+    ------
+    TypeError
+        If `cfg_developer` is not a Path-like object
     """
     if cfg_developer is None:
         cfg_developer = Path(__file__).parents[1] / 'config-developer.yml'
+    elif not isinstance(cfg_developer, Path):
+        raise TypeError("cfg_developer is not a Path-like object, got ",
+                        cfg_developer)
     mtime = cfg_developer.stat().st_mtime
```

Comment:
maybe just do
```suggestion
    mtime = Path(cfg_developer).stat().st_mtime
```
instead of the above? Or do you prefer the specific error message?

Reply:
I like the specific catch - raise that I put in - a bit more friendly for users since it prints out what they are passing in, also we  may want to tailor that in the future and accept other types of config developer?

Reply:
> we may want to tailor that in the future and accept other types of config developer?

Let's not do that, that would be awful ;-)